### ISSUE TITLE
structure expression of Chameleon is not supported

### ISSUE THREAD

**Issue by gforcada** [zopefoundation/z3c.pt#17]
Chameleon provides some [language extensions](https://chameleon.readthedocs.io/en/latest/reference.html#language-extensions), one of them is to use ``${structure: view.foo}`` rather than the more lengthy ``<div tal:replace="structure view/foo">``.

Seems though, that the integration here in Zope is missing that extension :confused:

**Comment by petschki** [comment]
Heres a duplicated issue https://github.com/zopefoundation/z3c.pt/issues/17 ... do not know where this actually belongs now and which one can be closed in favor of the main one ...

**Comment by d-maurer** [comment]
Peter Mathis wrote at 2023-6-30 00:28 -0700:
>Heres a duplicated issue https://github.com/zopefoundation/z3c.pt/issues/17 ... do not know where this actually belongs now and which one can be closed in favor of the main one ...

It depends:
as you know, `chameleon` lacks path expressions (because many frameworks
have no path concept or the concept is very different).
Therefore, `Zope` contains a `chameleon` integration -- implementing
among others `path` expressions in a `Zope` compatible way.
It also uses `zope.tales` (instead of `chameleon.tales`) as
this is also used widely in the `Zope` ecosystem (`CMFCore`,
`DCWorkflow`, Plone portlets, ...).

`structure` expression support requires integration of the expression
and template engines. Its implementation should be where those
engines are combined. For `Zope` this would be `Products.PageTemplates`
(part of `Zope`).


**If** `chameleon` exposes the `structure` wrapper, the implementation
should be simple.

**Comment by malthe** [comment]
I haven't looked at the details, but it shouldn't be a problem to add `structure` to the expression types.
```python
from chameleon.tales import StructureExpr
```
In Chameleon proper, the expression types are defined on the template class:
```python
class PageTemplate(BaseTemplate):
    ...
    expression_types = {
        'python': PythonExpr,
        'string': StringExpr,
        'not': NotExpr,
        'exists': ExistsExpr,
        'import': ImportExpr,
        'structure': StructureExpr,
    }
```

**Comment by icemac** [comment]
Fixed by #27.
